Sessions admin tests fail due to missing SENSITIVE_PATTERNS export

While running auth tests during the 1.2.3 security sweep (PR #1739), `packages/api/src/lib/auth/__tests__/sessions.test.ts` surfaces this error on main:

\`\`\`
[ERROR] Failed to load semantic improve routes — semantic improvement will be unavailable
  component: "admin-routes"
  err: SyntaxError: Export named 'SENSITIVE_PATTERNS' not found in module
       '/home/msywu/oss/atlas/ide/packages/api/src/lib/security.ts'.
\`\`\`

The admin semantic-improve routes try to import \`SENSITIVE_PATTERNS\` from \`packages/api/src/lib/security.ts\` but the export is missing. As a downstream effect, one test fails:

\`\`\`
(fail) Admin session routes > GET /api/v1/admin/sessions > returns sessions list with pagination
  Expected: "sess-1"
  Received: undefined
\`\`\`

Confirmed pre-existing (reproduced with \`git stash\` on a clean main checkout). Not caused by #1732/#1731 fix work, but should be tracked so the sessions admin page doesn't silently break in prod. Either:

1. Add the missing \`SENSITIVE_PATTERNS\` export to \`packages/api/src/lib/security.ts\`, or
2. Update the semantic-improve route to not require it, or
3. Fail loudly at startup rather than a warn.

Reproduce: \`bun test packages/api/src/lib/auth/__tests__/sessions.test.ts\`

Related: 1.2.3 security sweep PR #1739, tracker #1718

--- a/packages/api/src/api/routes/admin.ts
+++ b/packages/api/src/api/routes/admin.ts
@@ -264,7 +264,10 @@ admin.route("/cache", adminCache);
 admin.route("/cache/", adminCache);
 admin.route("/admin-actions", adminActions);
 admin.route("/admin-actions/", adminActions);
-// Plugin marketplace — lazy import to avoid crashing all admin routes if marketplace module fails
+// Plugin marketplace — dynamic import defers loading the marketplace module
+// (and its dependency graph) until admin routes register. Import failure is a
+// build/test bug, not a runtime-recoverable condition: log then re-throw so it
+// fails loudly instead of serving silent 404s on marketplace endpoints.
 try {
   const { workspaceMarketplace } = await import("./admin-marketplace");
   admin.route("/plugins/marketplace", workspaceMarketplace);
@@ -272,12 +275,14 @@ try {
 } catch (err) {
   log.error(
     { err: err instanceof Error ? err : new Error(String(err)) },
-    "Failed to load marketplace routes — plugin marketplace will be unavailable",
+    "Failed to load marketplace routes",
   );
+  throw err;
 }
 
-// Semantic improve routes — lazy import to avoid loading expert agent tools
-// (and their deep dependency graph) at admin module init time.
+// Semantic improve routes — dynamic import defers loading the expert agent tool
+// graph until admin routes register. Import failure fails loud for the same
+// reason as the marketplace import above.
 try {
   const { adminSemanticImprove } = await import("./admin-semantic-improve");
   admin.route("/semantic-improve", adminSemanticImprove);
@@ -285,8 +290,9 @@ try {
 } catch (err) {
   log.error(
     { err: err instanceof Error ? err : new Error(String(err)) },
-    "Failed to load semantic improve routes — semantic improvement will be unavailable",
+    "Failed to load semantic improve routes",
   );
+  throw err;
 }
 
 // Semantic entity editor routes — registered directly (not subrouter) to avoid

--- a/packages/api/src/lib/auth/__tests__/sessions.test.ts
+++ b/packages/api/src/lib/auth/__tests__/sessions.test.ts
@@ -207,6 +207,10 @@ mock.module("@atlas/api/lib/settings", () => ({
 
 mock.module("@atlas/api/lib/security", () => ({
   maskConnectionUrl: (_url: string) => "***",
+  // Must mirror every named export from lib/security.ts — partial mocks
+  // cause SyntaxError when admin.ts dynamically imports routes that
+  // transitively pull SENSITIVE_PATTERNS (e.g. admin-semantic-improve → sql tool).
+  SENSITIVE_PATTERNS: /__never_matches__/,
 }));
 
 mock.module("@atlas/api/lib/auth/server", () => ({

